### PR TITLE
bufferByCountAndTime

### PR DESCRIPTION
**Pull request by eulerfx**

### CODE REVIEW

File: tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs [hunk 1]
```diff
@@ -205,6 +205,27 @@ let ``AsyncSeq.bufferByCount empty``() =
   let s' = s |> AsyncSeq.bufferByCount 2 |> AsyncSeq.toList 
   Assert.True(([] = s'))
 
+
+[<Test>]
+let ``AsyncSeq.bufferByTimeAndCount``() =      
+  let s = asyncSeq {
+    yield 1
+    yield 2
```
Comment (dsyme) [inline tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs]: Great work. Could you also please add a more systematic test that covers a wider range of input sizes and buffer sizes?  e.g. compare to the tests I've been adding recently which always have some kind of "for" loop at the outer to cover a range of scenarios.  thanks

Comment (eulerfx) [inline tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs]: Will do.

Comment (eulerfx) [inline tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs]: Do you think FsCheck would make sense?

File: src/FSharp.Control.AsyncSeq/AsyncSeq.fs [hunk 2]
```diff
@@ -986,6 +986,44 @@ module AsyncSeq =
       if (buffer.Count > 0) then 
           yield buffer.ToArray() }
 
+  let bufferByCountAndTime (bufferSize:int) (timeoutMs:int) (source:AsyncSeq<'T>) : AsyncSeq<'T[]> = 
+    if (bufferSize < 1) then invalidArg "bufferSize" "must be positive"
+    if (timeoutMs < 1) then invalidArg "timeoutMs" "must be positive"
+    asyncSeq {
+      let buffer = new ResizeArray<_>()
+      use ie = source.GetEnumerator()
+      let rec loop rem rt = asyncSeq {
```
Comment (dsyme) [inline src/FSharp.Control.AsyncSeq/AsyncSeq.fs]: Most of the rest of the code uses "while" loops, for better or worse.  These use a Landin's-knot like trick to amortize the allocation that controls the loop. You may want to compare perf between these approaches at some point.

File: tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs (hunk 1 above)
Comment (dsyme) [inline tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs]: Yes, almost certainly would help, though I've not much experience in driving it

**Pull request merged**